src/confdb/converter/python/PythonConfigurationWriter.java

```diff
@@ -36,27 +37,42 @@
 public class PythonConfigurationWriter implements IConfigurationWriter {
 	protected ConverterEngine converterEngine = null;
 
+	/**
+	 * converts the configuration to python
+	 * note there are some thing not stored by the config which are added by this function
+	 * eg process.HLTConfigVersion, process.schedule, process.ProcessAcceleratorCUDA
+	 * these should all be listed in Configuration.reservedNames to ensure we dont accidently include
+	 * them in the config
+	 */
 	public String toString(IConfiguration conf, WriteProcess writeProcess) throws ConverterException {
-		String indent = "  ";
+		String indent = writeProcess == WriteProcess.YES ? "  " : "";
+		String object = writeProcess == WriteProcess.YES ? "process." : ""; 
 		StringBuffer str = new StringBuffer(100000);
+		StringBuffer strProcessLoads = new StringBuffer(100000);
 		// String fullName = conf.parentDir().name() + "/" + conf.name() + "/V" +
 		// conf.version() ;
 		String fullName = conf.toString();
 		str.append("# " + fullName + " (" + conf.releaseTag() + ")" + converterEngine.getNewline()
 				+ converterEngine.getNewline());
 
 		str.append("import FWCore.ParameterSet.Config as cms\n\n");
+
+		
 		if(conf.switchProducerCount() > 0) {
-		    str.append("from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA\n\n");
+			ReleaseVersionInfo relVarInfo = new ReleaseVersionInfo(conf.releaseTag());
+			str.append("from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA\n");
+			if( relVarInfo.cycle()>=13  ||
+				(relVarInfo.cycle()==12 && relVarInfo.major()>=3)){
```

Comment:
Does this check for 12_3_X also need a requirement like `prerel() == null`, to start support only from `12_3_0`? (`ProcessAcceleratorCUDA` was added in `CMSSW_12_3_0_pre6`)

Reply:
yes I'll update that

Reply:
I re-worked this to now check for >=CMSSW_12_3_0_pre6 (and moved the comparison func to the class)